integration-cli/docker_cli_daemon_test.go

```diff
@@ -896,3 +896,44 @@ func TestDaemonwithwrongkey(t *testing.T) {
 	os.Remove("/etc/docker/key.json")
 	logDone("daemon - it should be failed to start daemon with wrong key")
 }
+
+func TestDaemonRestartKillWait(t *testing.T) {
+	d := NewDaemon(t)
+	if err := d.StartWithBusybox(); err != nil {
+		t.Fatalf("Could not start daemon with busybox: %v", err)
+	}
+	defer d.Stop()
+
+	out, err := d.Cmd("run", "-d", "busybox", "/bin/cat")
+	if err != nil {
+		t.Fatalf("Could not run /bin/cat: err=%v\n%s", err, out)
+	}
+	containerID := strings.TrimSpace(out)
+
+	if out, err := d.Cmd("kill", containerID); err != nil {
+		t.Fatalf("Could not kill %s: err=%v\n%s", containerID, err, out)
+	}
+
+	if err := d.Restart(); err != nil {
+		t.Fatalf("Could not restart daemon: %v", err)
+	}
+
+	errchan := make(chan error)
+	go func() {
+		if out, err := d.Cmd("wait", containerID); err != nil {
```

Comment:
@cpuguy83 maybe I'm missing something here, could you look at the test that I removed in this pr? I wrote in IRC yestarday and I asked and said ok daemon start, cont start, cont kill, daemon restart, cont wait